app/controllers/user_mentee_applications_controller.rb

```diff
@@ -61,4 +63,12 @@ def user_mentee_application_params
   def resume_params
     params.require(:user_mentee_application).permit(:resume, :resume_name)
   end
+
+  def send_notifications
+    UserMenteeApplication::ApplicationSubmissionNotification
```

Comment:
In the [best practices](https://github.com/excid3/noticed#-best-practices) for the noticed gem, it suggests defining these notifications on the model and leveraging callbacks. I think this would be a good candidate for moving these methods out of the controller and instead defining them on application model, invoking them with `after_create_commit`s.

This would also prevent issues with notifications being sent, even if the transaction doesn't succeed.

Reply:
I interpret that section of the README not as a recommendation to use callbacks for these notifications, but as a recommendation to specifically use `after_create_commit` when you *are* using callbacks.

That said I think I might still be in favor of doing this via callback because having it in the controller means the notification side effects go untested.

Reply:
Lol, josh gets the careful reading award 😊  I actually agree that I don't think they should be in callbacks every time, but I'll advocate for it here still both for the testing reason as well as because the controllers may send them out despite an application failing to get created.

Reply:
done! good point @afogel.